src/components/profiles-modals/DeleteProfileModal.vue

```diff
@@ -0,0 +1,93 @@
+<script lang="ts">
+import { Vue, Component } from 'vue-property-decorator';
+import { ModalCard } from "../all";
+import FileUtils from "../../utils/FileUtils";
+import R2Error from "../../model/errors/R2Error";
+import Profile from "../../model/Profile";
+import FsProvider from "../../providers/generic/file/FsProvider";
+import path from "path";
+
+
+let fs: FsProvider;
+
+@Component({
+    components: {ModalCard}
+})
+export default class DeleteProfileModal extends Vue {
+
+    //TODO: add to vuex store or something to update Profiles list correctly.
```

Comment:
Also, this is one of those "getting sidetracked" things we discussed, so I would maybe recommend putting this branch on ice and starting by creating a PR that only contains the "move profile list to Vuex" change using the existing modals/implementation. Once that's done continuing on this work will be easier and a smaller PR.